bool stop
---
int64 succeed